server/data/data-jmap/src/test/java/org/apache/james/jmap/api/vacation/AbstractVacationRepositoryTest.java

```diff
@@ -59,18 +60,302 @@ public void retrieveVacationShouldReturnDefaultValueByDefault() {
+    public void modifyVacationShouldUpdateFromDate() {
+        VacationUpdate vacationUpdate = VacationUpdate.builder()
+            .fromDate(MissableOptional.of(DATE_2014))
+            .build();
+
+        vacationRepository.modifyVacation(ACCOUNT_ID, vacationUpdate).join();
+
+        assertThat(vacationRepository.retrieveVacation(ACCOUNT_ID).join())
+            .isEqualTo(Vacation.builder()
+                .fromDate(Optional.of(DATE_2014))
+                .enabled(false)
+                .build());
+    }
+
+    @Test
+    public void modifyVacationShouldUpdateToDate() {
+        VacationUpdate vacationUpdate = VacationUpdate.builder()
+            .toDate(MissableOptional.of(DATE_2017))
+            .build();
+
+        vacationRepository.modifyVacation(ACCOUNT_ID, vacationUpdate).join();
+
+        assertThat(vacationRepository.retrieveVacation(ACCOUNT_ID).join())
+            .isEqualTo(Vacation.builder()
+                .toDate(Optional.of(DATE_2017))
+                .enabled(false)
+                .build());
+    }
+
+    @Test
+    public void modifyVacationShouldUpdateSubject() {
+        String newSubject = "new subject";
+        VacationUpdate vacationUpdate = VacationUpdate.builder()
+            .subject(MissableOptional.of(newSubject))
+            .build();
+
+        vacationRepository.modifyVacation(ACCOUNT_ID, vacationUpdate).join();
+
+        assertThat(vacationRepository.retrieveVacation(ACCOUNT_ID).join())
+            .isEqualTo(Vacation.builder()
+                .subject(Optional.of(newSubject))
+                .enabled(false)
+                .build());
+    }
+
+    @Test
+    public void modifyVacationShouldUpdateTextBody() {
+        String newTextBody = "new text body";
+        VacationUpdate vacationUpdate = VacationUpdate.builder()
+            .textBody(MissableOptional.of(newTextBody))
+            .build();
+
+        vacationRepository.modifyVacation(ACCOUNT_ID, vacationUpdate).join();
+
+        assertThat(vacationRepository.retrieveVacation(ACCOUNT_ID).join())
+            .isEqualTo(Vacation.builder()
+                .textBody(newTextBody)
+                .enabled(false)
+                .build());
+    }
+
+    @Test
+    public void modifyVacationShouldUpdateHtmlBody() {
+        String newHtmlBody = "new <b>html</b> body";
+        VacationUpdate vacationUpdate = VacationUpdate.builder()
+            .htmlBody(MissableOptional.of(newHtmlBody))
+            .build();
+
+        vacationRepository.modifyVacation(ACCOUNT_ID, vacationUpdate).join();
+
+        assertThat(vacationRepository.retrieveVacation(ACCOUNT_ID).join())
+            .isEqualTo(Vacation.builder()
+                .enabled(false)
+                .htmlBody(newHtmlBody)
+                .build());
+    }
+
+    @Test
+    public void modifyVacationShouldAllowToUpdateAllFieldsAtOnce() {
+        VacationUpdate vacationUpdate = VacationUpdate.builder()
+            .subject(MissableOptional.ofOptional(VACATION.getSubject()))
```

Comment:
there's a lot of repetition, could we have a `VacationUpdate.builderFrom()` method ?